File: mall-sellergoods-interface/src/main/java/com/marshal/sellergoods/service/ItemCatService.java
package com.marshal.sellergoods.service;


import com.marshal.pojo.TbItemCat;
import com.marshal.util.ResponseData;
import java.util.List;

public interface ItemCatService {
	public List<TbItemCat> findAll();

	public ResponseData query(TbItemCat condition,int pageNum,int pageSize);

	public TbItemCat queryById(Long id);

	public void save(TbItemCat tbItemCat);

	public void delete(Long[] ids);
}

File: mall-manager-web/src/main/webapp/js/service/itemCatService.js
// service层,与数据库交互
app.service("itemCatService",function($http){
    //BASE crud
    this.query=function(pageNum,pageSize,condition){
        return $http.post("../itemCat/query?pageNum="+pageNum+"&pageSize="+pageSize,condition);
    }

    this.save=function (itemCat) {
        return $http.post("../itemCat/save",itemCat);
    }

    this.remove=function (idList) {
        return $http.get("../itemCat/delete?idList="+idList);
    }

    this.queryById=function (id) {
        return $http.get("../itemCat/queryById?id="+id);
    }
});